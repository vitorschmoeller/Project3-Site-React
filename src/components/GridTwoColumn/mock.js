export default {
  title: 'Grid Two Column',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Officia ullam facilis necessitatibus architecto modi doloribus pariatur minima enim consequuntur reiciendis dolore ipsam atque quae laboriosam, non consectetur sequi odio similique!`,
  srcImg: 'assets/images/javascript.svg',
};
