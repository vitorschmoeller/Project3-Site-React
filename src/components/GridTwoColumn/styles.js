import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
  display: grid;
  grid-template-columns: 1fr 2fr;
  align-items: center;
  gap: ${theme.spacings.large};

  @media ${theme.media.lteMedium} {
    grid-template-columns: 1fr;
    text-align: center;
  }
  ${Title} {

    margin-bottom: ${theme.spacings.xlarge};
  }
  `}


`;

export const TextContainer = styled.div`
  ${({ theme }) => css`


  @media (max-width: 768px) {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImgContainer = styled.div`

`;

export const Image = styled.img``;
