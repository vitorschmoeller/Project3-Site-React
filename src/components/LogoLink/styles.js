import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${() => css`
    display: flex;
    text-decoration: none;
    align-items: center;
    /* inherits diz que vai herdar do component pai */
    color: inherit;

    > img {
      max-height: 3rem;
    }
  `}
`;
