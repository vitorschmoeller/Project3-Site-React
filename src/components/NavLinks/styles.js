import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`

    display: flex;
    flex-flow: row wrap;
    font-size: ${theme.font.sizes.medium};

    @media ${theme.media.lteMedium}{
      flex-flow: column wrap;
      align-content: center;
    }

    /* @media (max-width: 768px) {
      display: block;
      text-align: center;
    } */
    `}
`;
