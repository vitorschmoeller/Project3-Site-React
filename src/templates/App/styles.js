import styled from 'styled-components';

export const Wrapper = styled.div`
  background-color: ${(props) => props.theme.colors.blue};

  h1 {
    background: red;

  }
`;
