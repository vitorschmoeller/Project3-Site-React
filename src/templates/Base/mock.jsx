import linksMock from '../../components/NavLinks/mock';

import gridMock from '../../components/GridSection/mock';
import { GridText } from '../../components/GridSection';

export const mockBase = {
  children: (
    <>
      <GridText {...gridMock} background />
      <GridText {...gridMock} />
      <GridText {...gridMock} background />
      <GridText {...gridMock} />
      <GridText {...gridMock} background />
      <GridText {...gridMock} />
    </>
  ),
  links: linksMock,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>Teste de footer</p>',
};
