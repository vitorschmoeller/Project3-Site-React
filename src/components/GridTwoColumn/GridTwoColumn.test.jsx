import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { GridTwoColumn } from '.';
import mock from './mock';
describe('<GridTwoColumn />', () => {
  it('Should render GridTwoColumn', () => {
    const { container } = renderTheme(
      <GridTwoColumn {...mock}></GridTwoColumn>,
    );

    expect(
      screen.getByRole('heading', { name: 'Grid Two Column' }),
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
