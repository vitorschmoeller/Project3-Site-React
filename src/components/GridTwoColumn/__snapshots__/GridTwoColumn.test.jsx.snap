// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> Should render GridTwoColumn 1`] = `
.c5 {
  color: #0a1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4.0rem;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #ffffff;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        <button>
          Dark
        </button>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <h2
              class="c4 c5"
            >
              Grid Two Column
               
            </h2>
            <div
              class="c6"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Officia ullam facilis necessitatibus architecto modi doloribus pariatur minima enim consequuntur reiciendis dolore ipsam atque quae laboriosam, non consectetur sequi odio similique!
            </div>
          </div>
          <div
            class=""
          >
            <img
              alt="Grid Two Column"
              class=""
              src="assets/images/javascript.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
